src/model/User.ts

```diff
@@ -39,29 +40,43 @@ const UserSchema: Schema<User> = new Schema({
         type: String,
         required: [true, "Email is required"],
         unique: true,
-        match: [/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/, "Please use a valid email"]
+        match: [/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/, "Invalid email format"]
     },
     password: {
         type: String,
         required: [true, "Password is required"]
     },
     verifyCode: {
         type: String,
-        required: [true, "Verify code is required"]
+        required: [true, "Verification code is required"]
     },
     verifyCodeExpiry: {
         type: Date,
-        required: [true, "Verify code expiry is required"]
+        required: [true, "Verification code expiry is required"]
     },
     isVerified: {
         type: Boolean,
         default: false
     },
-    isAcceptingMessages: { // Consistent naming
+    isAcceptingMessages: {
         type: Boolean,
         default: true
     },
-    messages: [MessageSchema]
+    messages: [{ type: Schema.Types.ObjectId, ref: 'Message' }]
+});
+
+UserSchema.index({ email: 1 }, { unique: true });
+UserSchema.index({ username: 1 }, { unique: true });
+
+UserSchema.methods.isVerifyCodeValid = function () {
+    return this.verifyCodeExpiry > Date.now();
+};
+
+// Hash password before saving
+UserSchema.pre('save', async function (next) {
+    if (!this.isModified('password')) return next();
+    this.password = await bcrypt.hash(this.password, 10);
+    next();
```

Comment:
Password is already getting stored via hashing process... not need to hash it again

Reply:
Hashing will only occur when the password is modified.

Reply:
Hashing will only occur when the password is modified.

Reply:
Hashing will only occur when the password is modified.